feat: implement header

hitendreamworld opened the pull request:
- render title and X(close) button

hitendreamworld committed:
feat: implement header [ci skip]

- render title and X(close) button

hitendreamworld committed:
fix: zero height when header is not available [ci skip]

hitendreamworld committed:
feat: new property show header

hitendreamworld committed:
feat: add style for header

hitendreamworld committed:
feat: update README.md [ci skip]

ChiragMoradiya commented on dw-menu.js:
```diff
@@ -105,10 +145,24 @@ export class DwMenu extends DwCompositeDialog {
 
   connectedCallback() {
     this._setDialogConfig();
+    this._showHeader = Boolean(this.heading) || this.showClose;
```
@hitendreamworld computed property value should be set from `willUpdate` in general practice. So, please update this implementation

hitendreamworld replied:
DwCompositeDialog uses the old `lit-element` version and that library does not provide `willUpdate` lifecycle. So, I wrote a custom getter/setter for `heading` and `showClose`.

ChiragMoradiya commented on dw-menu.js:
```diff
@@ -20,6 +26,35 @@ export class DwMenu extends DwCompositeDialog {
+      }
+
+      dw-icon-button {
+        width: max-content;
```
I don't understand why this css is needed for dw-icon-button

ChiragMoradiya commented on dw-menu.js:
```diff
@@ -105,10 +145,24 @@ export class DwMenu extends DwCompositeDialog {
+        ? html`<dw-icon-button
+            icon="close"
+            dismiss
+            @click=${(e) => console.log("close: invoked", e)}
```
remove this `console.log` if not needed.

hitendreamworld committed:
feat: reworks

- refactor
- custom getter/setter for heading and showClose
- on close icon click
- remove unnecessary styles

hitendreamworld committed:
feat: intruction for future implementation

dw-devops commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@dreamworld/dw-menu/v/1.0.0)
- [GitHub release](https://github.com/DreamworldSolutions/dw-menu/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
